### ISSUE TITLE
Using "Strong Fishing Pole" without required fishing skill

### ISSUE THREAD

**Issue by syssneck** [azerothcore/azerothcore-wotlk#7957]
### What client do you play on?

enUS

### Faction

- [X] Alliance
- [ ] Horde

### Content Phase:

- [X] Generic
- [ ] 1-19
- [ ] 20-29
- [ ] 30-39
- [ ] 40-49
- [ ] 50-59

### Current Behaviour

Original report: https://github.com/chromiecraft/chromiecraft/issues/1770

I purchased a “strong fishing pole” right after learning fishing (skill = 1) and found that it requires a fishing skill of 10 to use. Then I purchased a “fishing pole”, equipped it, and attached a “shiny bauble” to it. So, with this fishing pole equipped, my fishing skill was 26. Now, I simply swapped the fishing pole (equipped and with shiny bauble applied) to the strong fishing pole. After the swap, my skill in the skill panel is 6 (1 plus the 5 provided by the strong fishing pole) but I was allowed to fish with the strong fishing pole.

### Expected Blizzlike Behaviour

Too complicated and I don’t even know what keywords to search for to see whether this is a bug in retail. But it does not make sense, does it? I should not be allowed to use the strong fishing pole if the skill requirement is only met with another enhanced fishing pole. Let me know if there are similar “tricks” in the game (equip one item in a slot to meet the requirements for using another item for the same slot and swap.)

### Steps to reproduce the problem

Repeat what I described in “Current Behaviour” session.

### AC rev. hash/commit

https://github.com/chromiecraft/azerothcore-wotlk/commit/7e2e6f8ee85f82f57a3c6e0f6f645b4c42292e7a

### Operating system

Windows 7

### Modules

- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-chromie-xp](https://github.com/azerothcore/mod-chromie-xp)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-eluna-lua-engine](https://github.com/azerothcore/mod-eluna-lua-engine)
- [mod-ip-tracker](https://github.com/azerothcore/mod-ip-tracker)
- [mod-low-level-arena](https://github.com/azerothcore/mod-low-level-arena)
- [mod-multi-client-check](https://github.com/azerothcore/mod-multi-client-check)
- [mod-pvp-titles](https://github.com/azerothcore/mod-pvp-titles)
- [mod-pvpstats-announcer](https://github.com/azerothcore/mod-pvpstats-announcer)
- [mod-queue-list-cache](https://github.com/azerothcore/mod-queue-list-cache)
- [mod-server-auto-shutdown](https://github.com/azerothcore/mod-server-auto-shutdown)
- [lua-carbon-copy](https://github.com/55Honey/Acore_CarbonCopy)
- [lua-custom-worldboss](https://github.com/55Honey/Acore_CustomWorldboss)
- [lua-level-up-reward](https://github.com/55Honey/Acore_LevelUpReward)
- [lua-recruit-a-friend](https://github.com/55Honey/Acore_RecruitAFriend)
- [lua-send-and-bind](https://github.com/55Honey/Acore_SendAndBind)
- [lua-temp-announcements](https://github.com/55Honey/Acore_TempAnnouncements)
- [lua-zonecheck](https://github.com/55Honey/acore_Zonecheck)

### Customizations

None

### Server

ChromieCraft

Tester confirmed on local AC (606ea325a2eaf79a).
Maybe this trick (described in the original report) also works with other items.
When equipping items, there is only a check for the required skill when equipping, but not after the item was equipped, even if the required value is not met afterwards.
Wearing and using a Strong Fishing Pole with a fishing skill of 7 (+5 from the pole):
![strong fishing pole](https://user-images.githubusercontent.com/68868567/133890247-f9e61542-aa8e-4d9a-b894-58da229ee6dd.jpg)

**Comment by discorder0** [comment]
I think it is not about adding a check after equipping the strong fishing pole (or other item). The check should be done before equipping. The question is whether it should be done before (when fishing pole + shiny bauble is equipped) or IMMEDIATELY before (a moment when fishing pole + shiny bauble is unequipped to make room for strong fishing pole but the latter is not equipped yet) equipping the strong fishing pole.

If I were designing this game, the check will be immediately before the equipping. But not sure what is blizzlike.

I agree with @syssneck that this bug may have bigger impact if similar mechanism is affecting other items.

BTW, I’m the original reporter. Not sure whether I should chime in here or post my comments under my original report in chromiecraft (and @ the tester). Let me know.

**Comment by syssneck** [comment]
@discorder0 thanks for your comment! It's totally fine to post here!
And you are right, there shouldn't be a check, after the item was equipped.

Something just came to mind:
The check, that the requirements for a specific item are met, should also check, whether the item that is currently equipped on that slot, has stats from the same type of the requirement for the new item (in our case fishing). 
In this case, the stat gain from the current item should be subtracted from our overall skill (here fishing) for the follow-up check with the requirements for the new item, that we want to equip. (I'm not sure if this is clear enough)

Or even simpler, do the requirements check, as if you had no item in that slot equipped whatsoever (not sure if this brings other problems with it or if that is blizzlike).

I'm taking low priority out, as I'm not sure, if other items are affected as well. Thank you!

**Comment by discorder0** [comment]
@syssneck, you are right, there should be a check assuming the item that is currently in the slot has been removed. Removing this item may not only make the replacement not equippable, theoretically, it may also make items in other slots not functional.

**Comment by discorder0** [comment]
> Or even simpler, do the requirements check, as if you had no item in that slot equipped whatsoever (not sure if this brings other problems with it or if that is blizzlike at all).

I agree with you. Removing current item may not only make the replacement not equippable, theoretically it may also make items in other slots not functional.

**Comment by UltraNix** [comment]
The check is done before equipping the second item, but also before unequipping the first one, so core thinks you have higher skill.
